src/main/scala/com/shuttleql/services/game/matchmaking/MatchMaker.scala

```diff
@@ -83,23 +85,30 @@ object MatchMaker {
       }.values.flatten
 
     // 4. Find random split index
-    val randInd = Random.nextInt(numCourts) * 4
-
-    // 5. create teams after splitting by index
-    val (playerList1, playerList2) = randomPlayerList.splitAt(randInd)
-    matches = (playerList1 ++ playerList2)
-        .grouped(4)
-        .map { players => players.splitAt(2) }
-        .zipWithIndex
-        .map { case ((team1, team2), ind) =>
-          val courtId = ind + 1
-          Match(
-            team1 = team1.toList,
-            team2 = team2.toList,
-            courtName = "Court " + courtId,
-            courtId = courtId
-          )
-        }.toList
+    val splitIndex = Random.nextInt(randomPlayerList.size)
+
+    // 5. Create chunks of size 4 and size 2
+    val chunks = randomPlayerList.splitAt(splitIndex) match {
+      case (first, second) => {
+        val allChunks = (first.grouped(4).toList ::: second.toList.reverse.grouped(4).toList)
+        val (left, right) = (allChunks.filter(_.size == 4) ::: allChunks.filterNot(_.size == 4).flatten.grouped(4).toList).splitAt(numDoublesCourts)
+        left ::: right.flatten.grouped(2).toList
+      }
+    }
```

Comment:
Does this handle splitIndex > randomPlayerList.size? Do we need to fallback to empty array